crates/sovereign-sdk/full-node/db/sov-db/src/ledger_db/mod.rs

```diff
@@ -349,9 +349,13 @@ impl SharedLedgerOps for LedgerDB {
         match commitments {
             // If there were other commitments, upsert
             Some(mut commitments) => {
-                commitments.push(commitment);
-                self.db
-                    .put::<CommitmentsByNumber>(&SlotNumber(height), &commitments)
+                if !commitments.contains(&commitment) {
```

Comment:
I couldn't find any other way to fix duplicates in db. See https://github.com/chainwayxyz/citrea/actions/runs/12084846708/job/33700913239#step:13:2544

Reply:
the best way to do this would be to actually store commitments as with key `{slot_height}-{comm_start}-{comm_end}`, and then use the prefix iterator functionality of rocksdb to iterate over slot heights, which would ensure uniqueness of commitments in db level and would be still quite efficient since the keys will be ordered by nature.